deepspeed/runtime/zero/utils.py

```diff
@@ -158,3 +159,36 @@ def apply_to_tensors_only(function, value, warning_msg_fn=None):
                 logger.warning(warning_msg_fn(value))
                 warned = True
         return value
+
+
+@contextmanager
+def unwrap_model_for_generation(model):
```

Comment:
It is better to use general naming that describes what the utility does rather than specific usage like generation. 

```suggestion
def unshard_and_remove_hooks(model):
```